src/membraneWebRTC.ts

```diff
@@ -463,12 +495,13 @@ export class MembraneWebRTC {
         );
 
         this.callbacks.onTracksPriorityChanged?.(enabledTracks, disabledTracks);
+
       case "encodingSwitched":
-        this.callbacks.onTrackEncodingChanged?.(
-          deserializedMediaEvent.data.peerId,
-          deserializedMediaEvent.data.trackId,
-          deserializedMediaEvent.data.encoding
-        );
+        const trackId = deserializedMediaEvent.data.trackId;
+        const trackContext = this.trackIdToTrack.get(trackId)!;
+        trackContext.encoding = deserializedMediaEvent.data.encoding;
+        trackContext.encodingReason = deserializedMediaEvent.data.reason;
+        this.callbacks.onTrackEncodingChanged?.(trackContext);
```

Comment:
This way, you allow the user to destroy the internal state of our SDK from the callback defined on said SDK. If you want to pass trackContext as an argument for the callback, you will need to make a copy. A deep copy too, to avoid passing nested objects as reference as well.